src/TeaAsAService/Tea.Core/Domain/User.cs

```diff
@@ -35,15 +36,18 @@ public History CreateHistoryEntry()
             return entry;
         }
 
-        public void SetPassword(string newPassword)
+        public bool SetPassword(string newPassword)
         {
-            if (ValidatePassword(newPassword))
+            
+            if(newPassword.ValidatePassword())
             {
                 Password = newPassword;
+                return true;
             }
             else

```

Comment:
I would remove this else block and just return false at the bottom. No need to update the password to null either